Add Document controller documentation (#32)
kuzzleio/sdk-csharp: doc/1/controllers/document/m-replace/index.md

---
code: true
type: page
title: MReplaceAsync
description: Replaces documents.
---

# MReplaceAsync

Replaces multiple documents.

> Throws a partial error (error code 206) if one or more documents can not be replaced.

Reviewer: ```suggestion
Throws a partial error (error code 206) if one or more documents cannot be replaced.
```

## Arguments

```csharp
public async Task<JArray> MReplaceAsync(
  string index,
  string collection,
  JArray documents,
  bool waitForRefresh = false);

```

<br/>

| Argument     | Type                                 | Description                                       |
| ------------ | ------------------------------------ | ------------------------------------------------- |
| `index`      | <pre>string</pre>        | Index name                                        |
| `collection` | <pre>string</pre>        | Collection name                                   |

> | `documents`  | <pre>JArray</pre>        | JObject representing the documents to replace |

Reviewer: ```suggestion
| `documents`  | <pre>JArray</pre>        | Array of JObject instances, each representing a document to replace |
```

| `waitForRefresh`   | <pre>bool</pre><br/>(`false`)       | If `true`, waits for the change to be reflected for `search` (up to 1s)           |

### documents

> Each document has the following properties:

Reviewer: ```suggestion
Array of documents, each one with the following expected properties:
```

| Property  | Type              | Description                                            |
| --------- | ----------------- | ------------------------------------------------------ |
| `_id`      | <pre>string</pre> | Document ID      |

> | `body` | <pre>JObject</pre> | Document body |

Reviewer: (nitpicking)

```suggestion
| `body` | <pre>JObject</pre> | Document content |
```
